From 0ec8a0938fe9eaa1daa883dcba57c9c5f66a6c2f Mon Sep 17 00:00:00 2001
From: "Zheng, Xiao" <xiao.zheng@intel.com>
Date: Wed, 11 Mar 2015 13:58:04 +0800
Subject: [PATCH 003/403] vgt: Media code cleanup for BDW VCS2

1. _REG_VCS2_PP_DCLV and _REG_VCS2_PP_DIR_BASE are removed, since BDW has no
such registers.
2. change cmd: 3DSTATE_RAST_MULTISAMPLE to HSW only
3. add new BDW cmd: VEB_DI_IECP cmd.

Signed-off-by: Zheng, Xiao <xiao.zheng@intel.com>
---
 drivers/gpu/drm/i915/vgt/cmd_parser.c |    4 +++-
 drivers/gpu/drm/i915/vgt/handlers.c   |    7 -------
 drivers/gpu/drm/i915/vgt/reg.h        |    2 --
 3 files changed, 3 insertions(+), 10 deletions(-)

diff --git a/drivers/gpu/drm/i915/vgt/cmd_parser.c b/drivers/gpu/drm/i915/vgt/cmd_parser.c
index bfc035b..c1edb7f 100644
--- a/drivers/gpu/drm/i915/vgt/cmd_parser.c
+++ b/drivers/gpu/drm/i915/vgt/cmd_parser.c
@@ -1970,7 +1970,7 @@ static struct cmd_info cmd_info[] = {
 		0, 8, NULL},
 
 	{"3DSTATE_RAST_MULTISAMPLE", OP_3DSTATE_RAST_MULTISAMPLE, F_LEN_VAR, R_RCS,
-		D_GEN75PLUS, 0, 8, NULL},
+		D_HSW, 0, 8, NULL},
 
 	{"3DSTATE_STENCIL_BUFFER", OP_SNB_3DSTATE_STENCIL_BUFFER, F_LEN_VAR, R_RCS,
 		D_SNB, ADDR_FIX_1(2), 8, NULL},
@@ -2239,6 +2239,8 @@ static struct cmd_info cmd_info[] = {
 	{"VEBOX_SURFACE_STATE", OP_VEB_SURFACE_STATE, F_LEN_VAR, R_VECS, D_HSW_PLUS, 0, 12, NULL},
 
 	{"VEB_DI_IECP", OP_VEB_DNDI_IECP_STATE, F_LEN_VAR, R_VECS, D_HSW, 0, 12, NULL},
+
+	{"VEB_DI_IECP", OP_VEB_DNDI_IECP_STATE, F_LEN_VAR, R_VECS, D_BDW_PLUS, 0, 20, NULL},
 };
 
 static int cmd_hash_init(struct pgt_device *pdev)
diff --git a/drivers/gpu/drm/i915/vgt/handlers.c b/drivers/gpu/drm/i915/vgt/handlers.c
index fa4eeda..e4e4052 100644
--- a/drivers/gpu/drm/i915/vgt/handlers.c
+++ b/drivers/gpu/drm/i915/vgt/handlers.c
@@ -583,7 +583,6 @@ static int mmio_to_ring_id(unsigned int reg)
 	case _REG_VECS_EXECLIST_STATUS:
 		ring_id = RING_BUFFER_VECS;
 		break;
-	case _REG_VCS2_PP_DIR_BASE:
 	case _REG_VCS2_MFX_MODE_BDW:
 	case _REG_VCS2_EXECLIST_SUBMITPORT:
 	case _REG_VCS2_EXECLIST_STATUS:
@@ -2574,12 +2573,10 @@ reg_attr_t vgt_base_reg_info[] = {
 {_REG_RCS_PP_DIR_BASE_READ, 4, F_RDR_ADRFIX, 0xFFFFF000, D_SNB, NULL, NULL},
 {_REG_RCS_PP_DIR_BASE_IVB, 4, F_RDR_ADRFIX, 0xFFFFF000, D_PRE_BDW, NULL, NULL},
 {_REG_VCS_PP_DIR_BASE, 4, F_RDR_ADRFIX, 0xFFFFF000, D_PRE_BDW, NULL, NULL},
-{_REG_VCS2_PP_DIR_BASE, 4, F_RDR_ADRFIX, 0xFFFFF000, D_BDW_PLUS, NULL, NULL},
 {_REG_BCS_PP_DIR_BASE, 4, F_RDR_ADRFIX, 0xFFFFF000, D_PRE_BDW, NULL, NULL},
 {_REG_VECS_PP_DIR_BASE, 4, F_RDR_ADRFIX, 0xFFFFF000, D_HSW, NULL, NULL},
 {_REG_RCS_PP_DCLV, 4, F_RDR, 0, D_ALL, NULL, NULL},
 {_REG_VCS_PP_DCLV, 4, F_RDR, 0, D_ALL, NULL, NULL},
-{_REG_VCS2_PP_DCLV, 4, F_RDR, 0, D_BDW_PLUS, NULL, NULL},
 {_REG_BCS_PP_DCLV, 4, F_RDR, 0, D_ALL, NULL, NULL},
 {_REG_VECS_PP_DCLV, 4, F_RDR, 0, D_HSW, NULL, NULL},
 {_REG_RBSYNC, 4, F_RDR, 0, D_ALL, NULL, NULL},
@@ -3642,10 +3639,6 @@ bool vgt_post_setup_mmio_hooks(struct pgt_device *pdev)
 				ring_pp_mode_write);
 
 	if (IS_BDWGT3(pdev)) {
-		reg_update_handlers(pdev, _REG_VCS2_PP_DIR_BASE, 4,
-				pp_dir_base_read, pp_dir_base_write);
-		reg_update_handlers(pdev, _REG_VCS2_PP_DCLV, 4,
-				pp_dclv_read, pp_dclv_write);
 		reg_update_handlers(pdev, _REG_VCS2_MFX_MODE_BDW, 4,
 				ring_pp_mode_read,
 				ring_pp_mode_write);
diff --git a/drivers/gpu/drm/i915/vgt/reg.h b/drivers/gpu/drm/i915/vgt/reg.h
index 433ef8c..0c39a2a 100644
--- a/drivers/gpu/drm/i915/vgt/reg.h
+++ b/drivers/gpu/drm/i915/vgt/reg.h
@@ -257,8 +257,6 @@ static inline uint32_t __RING_REG(int32_t ring_id, uint32_t rcs_reg)
 #define _REG_BCS_PP_DCLV		0x22220
 #define _REG_VCS_PP_DIR_BASE		0x12228
 #define _REG_VCS_PP_DCLV		0x12220
-#define _REG_VCS2_PP_DIR_BASE		0x1c228
-#define _REG_VCS2_PP_DCLV		0x1c220
 #define _REG_VECS_PP_DIR_BASE		0x1A228
 #define _REG_VECS_PP_DCLV		0x1A220
 
-- 
1.7.10.4

